invenio_webhooks/models.py

```diff
@@ -157,9 +157,28 @@ class CeleryReceiver(Receiver):
     it synchronously during the request.
     """
 
+    CELERY_STATES_TO_HTTP = {
+        states.PENDING: 202,
+        states.STARTED: 202,
+        states.RETRY: 202,
+        states.FAILURE: 500,
+        states.SUCCESS: 201,
+    }
+    """Mapping of Celery result states to HTTP codes."""
+
     def __call__(self, event):
         """Fire a celery task."""
-        process_event.apply_async(args=[event.id])
+        process_event.apply_async(task_id=str(event.id), args=[event.id])
+
+    def status(self, event):
+        """Return current processing status."""
+        result = AsyncResult(str(event.id))
+        return self.CELERY_STATES_TO_HTTP.get(result.state)
```

Comment:
I will do some more investigation in next PR where the `meta` will be included in HTTP headers.